crates/diesel_models/src/merchant_account.rs

```diff
@@ -197,6 +197,57 @@ impl MerchantAccount {
     }
 }
 
+impl MerchantAccount {
+    #[cfg(feature = "v1")]
+    pub fn from_update(
+        &mut self,
+        update: MerchantAccountUpdateInternal,
+    ) -> Result<(), &'static str> {
+        macro_rules! update {
+            ([$(($attr:ident,)),*]) => {
+
+                $(self.$attr = update.$attr;)*
+            };
+            ([$(($attr:ident,$error:literal)),*])=>{
+                $(self.$attr = update.$attr.ok_or(concat!($error,stringify!($attr)))?;)*
```

Comment:
This would cause it to throw an error for all fields that are `None`?

That is not the intended behavior here, fields being `None` only signify that they would not be included in the `UPDATE` query, and thus would not be updated.

From [the diesel docs](https://diesel.rs/guides/all-about-updates.html):

> If the struct has any optional fields on it, these will also have special behavior. By default, `#[derive(AsChangeset)]` will assume that `None` means that you don’t wish to assign that field.